python/chemiscope/input.py

```diff
@@ -377,6 +382,54 @@ def create_input(
     return data
 
 
+def write_external_structures(frames, prefix="structure", compresslevel=5):
```

Comment:
Why are we writing to the disk? I was more thinking to implement this as a fully in-memory thing.

Either the data fits into Python memory (but not JS memory), and then we can load everything in Python, create "UserStructure" and send that to the JS side, while keeping the Python data for the frames around. Then when JS requests a frame, we can transform it to JSON and return just this one.

If the data does not even fit in Python memory, we could have a callback that fires when JS requests more data, reading the one frame we need from disk and converting it (although I'd say this is out of scope for this PR)

Reply:
I would first get something that can read from disk. Then if we want to add some caching python-side, it can be done easily in a separate PR. on the sphinx side, it serves the same purpose as the loader for the website, faster loading of a page.